### PR TITLE
OCP-37031: OVN handles projects that start with a digit

### PR DESCRIPTION
**Pull request by rbbratta**
BZ 1899746

### CODE REVIEW

**Comment by rbbratta** [comment]
http://file.rdu.redhat.com/~rbrattai/logs/cucushift-2020-11-24-00-01-53.log

File: features/networking/ovn.feature [hunk 1]
```diff
@@ -444,3 +446,21 @@ Feature: OVN related networking scenarios
...
+    When I run the :create client command with:
+      | f | list_for_pods.json |
+    Then the step should succeed
+    Given 2 pods become ready with labels:
+      | name=test-pods |
+    When I run command on the "<%= pod(0).node_name %>" node's sdn pod:
+      | bash | -c | ovs-vsctl list interface \| grep iface-id |
+    Then the step should succeed
+    # the iface-id should be quoted if it starts with a digit
+    And the output should contain "<%= project.name %>"
```
Comment (zhaozhanqi) [inline features/networking/ovn.feature:466]: hmm, seems here guarantee the project is quoted. how about " "<%= project.name %>"

File: features/step_definitions/project.rb [hunk 2]
```diff
@@ -66,6 +66,19 @@
   end
 end
 
+# create a new project w/ a leading digit to test OVS quoting
+Given /^I create a project with leading digit name$/ do
```
Comment (akostadinov) [inline features/step_definitions/project.rb:70]: You could also unify with the non-leading step like:
`I create a project with (non-)?leading digit name`

but it is fine to leave as is

File: features/networking/ovn.feature (hunk 1 above)
Comment (rbbratta) [reply features/networking/ovn.feature:466]: It looks like the current regex won't allow `"`  inside the `""`

```ruby
Then /^(the|all)? outputs?( by order)? should( not)? (contain|match) "(.+)"(?: (\d+) times)?$/ do |all, in_order, negative, match_type, pattern, times|
```

I was thinking about matching in the `grep iface-id`.  Not sure how directly we want to match the OVS output.

If OVN does fail the Pods should not become ready, but I haven't verified the failure case yet.

Comment (rbbratta) [reply features/networking/ovn.feature:466]: Yes, just tested on a build with the bug, the `    Given 2 pods become ready with labels:` step fails.  So the `ovs-vsctl` is just an extra debug type check to dump the `iface-id` I suppose.

Failure example
```
  Warning  FailedCreatePodSandBox  4m50s (x32 over 12m)  kubelet, v7c8g.c.openshift-qe.internal  (combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_test-rc-gw8tk_56784etn_5dae58b7-1678-426d-bedd-4b83966b8d01_0(05ee5ad713612e8eabf1626b0b8bab31c757734a3597ae8002ee4636bc14e633): [56784etn/test-rc-gw8tk:ovn-kubernetes]: error adding container to network "ovn-kubernetes": CNI request failed with status 400: '[56784etn/test-rc-gw8tk 05ee5ad713612e8eabf1626b0b8bab31c757734a3597ae8002ee4636bc14e633] [56784etn/test-rc-gw8tk 05ee5ad713612e8eabf1626b0b8bab31c757734a3597ae8002ee4636bc14e633] failed to configure pod interface: error while waiting on flows for pod: OVS sandbox port 05ee5ad713612e8 is no longer active (probably due to a subsequent CNI ADD)
'
[18:23:27] INFO> Exit Status: 0

RuntimeError: pod test-rc-gw8tk did not reach expected status
verification-tests/features/step_definitions/pod.rb:157:in `block (2 levels) in <top (required)>'
```

Comment (anuragthehatter) [reply features/networking/ovn.feature:466]: hmm..needs to be 
```
And the output should contain:
   | "<%= project.name %> |
```

Comment (zhaozhanqi) [reply features/networking/ovn.feature:466]: it's fine if the pod can works well.

> Yes, just tested on a build with the bug, the ` Given 2 pods become ready with labels:` step fails. So the `ovs-vsctl` is just an extra debug type check to dump the `iface-id` I suppose.
> 
> Failure example
> 
> ```
>   Warning  FailedCreatePodSandBox  4m50s (x32 over 12m)  kubelet, v7c8g.c.openshift-qe.internal  (combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_test-rc-gw8tk_56784etn_5dae58b7-1678-426d-bedd-4b83966b8d01_0(05ee5ad713612e8eabf1626b0b8bab31c757734a3597ae8002ee4636bc14e633): [56784etn/test-rc-gw8tk:ovn-kubernetes]: error adding container to network "ovn-kubernetes": CNI request failed with status 400: '[56784etn/test-rc-gw8tk 05ee5ad713612e8eabf1626b0b8bab31c757734a3597ae8002ee4636bc14e633] [56784etn/test-rc-gw8tk 05ee5ad713612e8eabf1626b0b8bab31c757734a3597ae8002ee4636bc14e633] failed to configure pod interface: error while waiting on flows for pod: OVS sandbox port 05ee5ad713612e8 is no longer active (probably due to a subsequent CNI ADD)
> '
> [18:23:27] INFO> Exit Status: 0
> 
> RuntimeError: pod test-rc-gw8tk did not reach expected status
> verification-tests/features/step_definitions/pod.rb:157:in `block (2 levels) in <top (required)>'
> ```

ok, if the pod is running can make sure. that's ok

**Pull request merged**